docs/design.md

```diff
@@ -220,6 +220,40 @@ and during development, as well as any desired safety margin on the
 manufacturer's specified number of erase cycles. In general, using a ratio that
 allows hundreds to thousands of field upgrades in production is recommended.
 
+### [Swap without using scratch](#image-swap-no-scratch)
+
+This algorithm is an alternative to the swap using the scratch algorithm.
+It uses an additional sector in the primary slot to make swap possible.
+The algorithm works as follows:
+
+  1.	Moves all sectors of the primary slot up by one sector.  
+    Beginning from N=0:  
+  2.	Copies the N-th sector from the secondary slot to the N-th sector of the
+  primary slot.
+  3.	Copies the (N+1)-th sector from the primary slot to the N-th sector of the
+  secondary slot.
+  4.	Repeats steps 2. and 3. until all the slots' sectors are swapped.
+
+This algorithm is designed so that the higher sector of the primary slot is
+used only for allowing sectors to move up. Therefore the most
+memory-size-effective slot layout is when the primary slot is exactly one sector
+larger than the secondary slot, although same-sized slots are allowed as well.
+The algorithm is limited to support sectors of the same
+sector layout. All slot's sectors should be of the same size.
+
+When using this algorithm the maximum image size available for the application
+will be:  
+maximum-image-size = (N-1) * slot-sector-size - image-trailer-size
```

Comment:
It's important to note that the subtraction is not by the `image-trailer-size` itself, but to a sector alignment that fits it, so it would most probably be 4K in 4K sector devices, but if the slot is really large it could be 8K (two sectors) or more; and only the other N-2 (or N-x) sectors would be moved up and swapped. This is in contrast with swap using scratch where the image can almost touch the trailer in bytes.